source/extensions/retry/host/omit_host_metadata/omit_host_metadata.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+class OmitHostsRetryPredicate : public Upstream::RetryHostPredicate {
+public:
+  explicit OmitHostsRetryPredicate(
+      const envoy::config::core::v3alpha::Metadata metadata_match_criteria)
+      : metadata_match_criteria_(metadata_match_criteria) {
+    const auto& filter_it = metadata_match_criteria_.filter_metadata().find(
+        Envoy::Config::MetadataFilters::get().ENVOY_LB);
+    if (filter_it == metadata_match_criteria_.filter_metadata().end()) {
+      throw EnvoyException("No metadata match criteria defined.");
```

Comment:
Looking through the code a bit it looks like this would actually end up being handled on the request path, so I don't think we should be throwing an exception. We probably need to handle this case (probably just default to accepting all hosts). 

If we want to support these kind of validations we probably need to add a validation method to the factory so that it can be validated on configuration time - the predicates themselves are created in the request path so that their lifetime is limited to a single request.

Reply:
So just to confirm, for the time being you would be okay to have this validation to move back to shouldSelectAnotherHost method and defaults to selecting all hosts (as it was earlier)?